langchain4j-astradb/src/main/java/dev/langchain4j/store/memory/chat/astradb/AstraDbChatMessage.java

```diff
@@ -0,0 +1,161 @@
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.ChatMessageType;
+import dev.langchain4j.data.message.Content;
+import dev.langchain4j.data.message.SystemMessage;
+import dev.langchain4j.data.message.UserMessage;
+import lombok.Data;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A chat message stored in AstraDB.
+ */
+@Setter @Getter @NoArgsConstructor
+public class AstraDbChatMessage {
+
+    /** Public Static to help build filters if any. */
+    public static final String PROP_CHAT_ID = "chat_id";
+
+    /** Public Static to help build filters if any. */
+    public static final String PROP_MESSAGE = "text";
+
+    /** Public Static to help build filters if any. */
+    public static final String PROP_MESSAGE_TIME = "message_time";
+
+    /** Public Static to help build filters if any. */
+    public static final String PROP_MESSAGE_TYPE = "message_type";
+
+    /** Public Static to help build filters if any. */
+    public static final String PROP_TOOLS = "tools_execution_requests";
+
+    /** Public Static to help build filters if any. */
+    public static final String PROP_CONTENTS = "contents";
+
+    /** Public Static to help build filters if any. */
+    public static final String PROP_NAME = "name";
+
+    @JsonProperty(PROP_CHAT_ID)
+    private String chatId;
+
+    @JsonProperty(PROP_MESSAGE_TYPE)
+    private ChatMessageType messageType;
+
+    @JsonProperty(PROP_MESSAGE_TIME)
+    private Instant messageTime;
+
+    @JsonProperty(PROP_NAME)
+    private String name;
+
+    @JsonProperty(PROP_MESSAGE)
+    private String text;
+
+    @JsonProperty(PROP_TOOLS)
+    private List<ToolExecutionRequest> toolExecutionRequests;
+
+    @JsonProperty(PROP_CONTENTS)
+    private List<AstraDbContent> contents;
+
+    @Data @NoArgsConstructor
+    public static class ToolExecutionRequest {
+        private String id;
+        private String name;
+        private String arguments;
+        public ToolExecutionRequest(dev.langchain4j.agent.tool.ToolExecutionRequest lc4jTER) {
+            this.id = lc4jTER.id();
+            this.name = lc4jTER.name();
+            this.arguments = lc4jTER.arguments();
+        }
+        public dev.langchain4j.agent.tool.ToolExecutionRequest asLc4j() {
+            return dev.langchain4j.agent.tool.ToolExecutionRequest.builder()
+                    .id(this.id)
+                    .name(this.name)
+                    .arguments(this.arguments)
+                    .build();
+        }
+    }
+
+    /**
+     * Constructor using the Langchain4J abstraction
+     * @param chatMessage
+     *      lc4j classes
+     */
+    public AstraDbChatMessage(ChatMessage chatMessage) {
```

Comment:
Just in case, you could use `ChatMessageSerializer` and `ChatMessageDeserializer` to easily serialize and deserialize all message types to/from JSON